### PR TITLE
scatter_5

### LINKED ISSUE #164: scatter_5
Description: Grouping data by value and plotting with different colors and markers

Script URL: https://www.ncl.ucar.edu/Applications/Scripts/scatter_5.ncl
Plot URL: https://www.ncl.ucar.edu/Applications/Images/scatter_5_lg.png

Category: Scatter

### PR DESCRIPTION
**Pull request by hCraker**
Closes #164 
image

### CODE REVIEW

File: Plots/Scatter/NCL_scatter_5.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+# Divide data into 8 bins and plot
+numBins = 8
+indices = np.arange(0, 300)
+partitions = np.linspace(0, 20, numBins + 1)
+label = "{start:g}:{end:g}"
+for x in range(0, numBins):
+    bins = np.where(data > partitions[x], data, np.nan)
+    bins = np.where(bins < partitions[x + 1], bins, np.nan)
+    indices = np.where(bins != np.nan, indices, np.nan)
+    ax.scatter(indices, bins, marker=markers[x],
```
Comment (clairefio) [inline Plots/Scatter/NCL_scatter_5.py:59]: I would add a fillstyle='none' argument here to make the markers look like the ones in the NCL graph

Comment (hCraker) [reply Plots/Scatter/NCL_scatter_5.py:59]: For some reason, `scatter` doesn't have that as a kwarg. I'll have to use `plt.plot` instead, but I think I can get it to work

Comment (hCraker) [reply Plots/Scatter/NCL_scatter_5.py]: Fixed!

**Comment by hCraker** [comment]
image

**Comment by michaelavs** [comment]
When I run the code locally, the labels for the points on the graph aren't showing up. This may just be something from my side not displaying correctly, so I'm going to past the plot here and see what happens. Is this happening on your side too @clairefio?

image

**Comment by hCraker** [comment]
> When I run the code locally, the labels for the points on the graph aren't showing up. This may just be something from my side not displaying correctly, so I'm going to past the plot here and see what happens. Is this happening on your side too @clairefio?
> 
> image

Thanks for pointing this out @michaelavs. What's odd is that the plot shows up as expected on Sphinx when I build the documentation locally. What does the documentation page look like when built on your machine?

**Comment by michaelavs** [comment]
> > When I run the code locally, the labels for the points on the graph aren't showing up. This may just be something from my side not displaying correctly, so I'm going to past the plot here and see what happens. Is this happening on your side too @clairefio?
> > image
> 
> Thanks for pointing this out @michaelavs. What's odd is that the plot shows up as expected on Sphinx when I build the documentation locally. What does the documentation page look like when built on your machine?

It does show up when I run sphinx, so definitely something just locally that was preventing it before!

**Comment by hCraker** [comment]
@jukent I wasn't able to figure out how to use xarray instead of numpy for this. The plot still looks the way it should, but it might not be the simplest implementation. Thoughts?

File: Plots/Scatter/NCL_scatter_5.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+##############################################################################
+# Import packages:
+import numpy as np
+import matplotlib.pyplot as plt
+from cycler import cycler
+
+from geocat.viz import util as gvutil
+
+##############################################################################
+# Generate random data with an average of 10 and a stddev of 3
```
Comment (erogluorhan) [inline Plots/Scatter/NCL_scatter_5.py:30]: Despite knowing that this comment is coming from NCL script, how about this for more clarity: "Generate random data from a normal (Gaussian) distribution with a mean of 10 and standard deviation of 3"

**Pull request merged**